pkg/ebpf/policy_manager.go

```diff
@@ -10,47 +10,129 @@ import (
 // policyManager is a thread-safe struct that manages the enabled policies for each rule
 type policyManager struct {
 	mutex sync.Mutex
-	rules map[events.ID]uint64
+	rules map[events.ID]*eventState
+}
+
+// eventState is a struct that holds the state of a given event
+type eventState struct {
+	policyMask uint64
+	enabled    bool
 }
 
 func newPolicyManager() *policyManager {
 	return &policyManager{
 		mutex: sync.Mutex{},
-		rules: make(map[events.ID]uint64),
+		rules: make(map[events.ID]*eventState),
+	}
+}
+
+// IsEnabled tests if a event, or a policy per event is enabled (in the future it will also check if a policy is enabled)
+// TODO: add metrics about an event being enabled/disabled, or a policy being enabled/disabled?
+func (pm *policyManager) IsEnabled(matchedPolicies uint64, ruleId events.ID) bool {
+	pm.mutex.Lock()
+	defer pm.mutex.Unlock()
+
+	if !pm.isEventEnabled(ruleId) {
+		return false
 	}
+
+	return pm.isRuleEnabled(matchedPolicies, ruleId)
 }
 
 // IsRuleEnabled returns true if a given event policy is enabled for a given rule
 func (pm *policyManager) IsRuleEnabled(matchedPolicies uint64, ruleId events.ID) bool {
 	pm.mutex.Lock()
 	defer pm.mutex.Unlock()
 
-	policyMask, ok := pm.rules[ruleId]
+	return pm.isRuleEnabled(matchedPolicies, ruleId)
+}
+
+// not synchronized, use IsRuleEnabled instead
+func (pm *policyManager) isRuleEnabled(matchedPolicies uint64, ruleId events.ID) bool {
```

Comment:
`isRuleEnabled()` is the unique without mutex, do you have plans to use this internally?

Reply:
Actually, there are two without mutex `isRuleEnabled` and `isEventEnabled`, because they are used in two places either `IsRuleEnabled` or `IsEventEnabled` and `IsEnabled`.

Reply:
Oh, they are, indeed. Thanks. 👍🏼